include/seqan3/alphabet/concept.hpp

```diff
@@ -43,219 +43,81 @@
-struct underlying_char
-{
-    //!\brief The alphabet's char_type.
-    using type = typename alphabet_type::char_type;
-};
-
-//!\brief Shortcut for seqan3::underlying_char
-template <typename alphabet_type>
-//!\relates seqan3::underlying_char
-using underlying_char_t = typename underlying_char<alphabet_type>::type;
-
-//!\brief Type metafunction that returns the `rank_type` defined inside an alphabet type.
-//!\tparam alphabet_type Must provide a `rank_type` member type.
-template <typename alphabet_type>
-//!\cond
-    requires requires (alphabet_type c) { typename alphabet_type::rank_type; }
-//!\endcond
-struct underlying_rank
-{
-    //!\brief The alphabet's rank_type.
-    using type = typename alphabet_type::rank_type;
-};
-
-//!\brief Shortcut for seqan3::underlying_rank
-//!\relates seqan3::underlying_rank
-template <typename alphabet_type>
-using underlying_rank_t = typename underlying_rank<alphabet_type>::type;
-
-// ------------------------------------------------------------------
-// value metafunctions
-// ------------------------------------------------------------------
-
-//!\brief Value metafunction that returns the `value_size` defined inside an alphabet type.
-//!\tparam alphabet_type Must provide a `value_size` static member.
-template <typename alphabet_type>
-//!\cond
-    requires requires (alphabet_type c) { alphabet_type::value_size; }
-//!\endcond
-struct alphabet_size
-{
-    //!\brief The alphabet's size.
-    static constexpr underlying_rank_t<alphabet_type> value = alphabet_type::value_size;
-};
-
-//!\brief Shortcut for seqan3::alphabet_size
-//!\relates seqan3::alphabet_size
-template <typename alphabet_type>
-constexpr underlying_rank_t<alphabet_type> alphabet_size_v = alphabet_size<alphabet_type>::value;
-
-// ------------------------------------------------------------------
-// free functions
-// ------------------------------------------------------------------
-
-/*!\name Free function wrappers for alphabet member functions
- * \brief For alphabets that implement needed functions as members, make them "globally" available.
- * \{
- */
-/*!\brief Free function wrapper that calls `.to_char()` on the argument.
- * \tparam alphabet_type Must provide a `.to_char()` member function.
- * \param c The alphabet letter that you wish to convert to char.
- * \returns The letter's value in the alphabet's rank type (usually `char`).
- */
-template <typename alphabet_type>
-constexpr underlying_char_t<alphabet_type> to_char(alphabet_type const c)
-    requires requires (alphabet_type c) { { c.to_char() } -> underlying_char_t<alphabet_type>; }
-{
-    return c.to_char();
-}
-
-/*!\brief Ostream operator that delegates to `c.to_char()`.
- * \tparam alphabet_type Must provide a `.to_char()` member function.
- * \param os The output stream you are printing to.
- * \param c The alphabet letter that you wish to convert to char.
- * \returns A reference to the output stream.
+/*!\interface seqan3::semi_alphabet_concept <>
```

Comment:
semi_* is a convention from the STL, see also http://en.cppreference.com/w/cpp/experimental/ranges/concepts/Semiregular

It also makes sense, because it has exactly half the functionality of an alphabet (all the rank stuff, but not the char stuff).